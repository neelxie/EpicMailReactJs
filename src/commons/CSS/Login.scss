.loginform {
    background-color: pink;
    margin: auto;
    width: 70%;
    height: 70vh;
    margin-top: 10vh;
    border-radius: 5px;
    padding: 20px;
    text-align: center;
    border: 15px 23px yellow solid;
    box-shadow: 100px white;
    display: block;

    .login-wording {
        font-family: cursive;
        font-size: 3.8rem;
        margin-top:3vh;
        margin-bottom: 55px;
    }

    input {
        display: block;
        margin: auto;
        margin-bottom: 45px;
        width: 50%;
        font-size: 1.5rem;
        border-radius: 13px;
        padding-left: 14px;

        &:hover {
            width: 60%;
        }
    }
    button {
        color: white;
        background-color:  rgb(70, 66, 248);
        width: 50%;
        font-size: 1.8rem;
        border-radius: 13px;

        &:hover {
            background-color: blue;
            width: 60%;
        }
    }
    .login-signup-link {
        font-size: 1.3rem;
        padding-top: 25px;

        a {
            color: blue;
            font-size: 1.35rem;

            &:hover {
                color: red;
                font-size: 1.4rem;
            }
        }
    }
}