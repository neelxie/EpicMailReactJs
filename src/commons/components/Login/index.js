/* eslint-disable import/no-extraneous-dependencies */
import React, { Component } from "react";
import { connect } from "react-redux";
import loginActions from "../../../redux/actions/loginActions";
import '../../CSS/Login.scss';
import { Link } from 'react-router-dom';

class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
    };
  }
  onChangeHandler = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  }

  
  onSubmitHandler = (e) => {
    e.preventDefault();

    const { email, password } = this.state;
    const { loggingIn } = this.props;

    // if (email && password) {
    loggingIn(email, password);
    // }
  }
  render() {
    const { email, password } = this.state;
    const { errorMessage, isProcessing, userMsg } = this.props;
    return (
      <div className='loginform'>
        <div className="login-wording">Login into Epic Mail</div>
        <form name='form' onSubmit={this.onSubmitHandler}>
          <input
            className="login-email"
            id="login-email"
            name="email"
            type="email"
            value={email}
            onChange={this.onChangeHandler}
            placeholder="derek@gmail.com"
            required
          />
          <input
            className="login-password"
            id="login-password"
            name="password"
            type="password"
            value={password}
            onChange={this.onChangeHandler}
            placeholder="password"
            required
          />
          {isProcessing && <span>is processing</span>}
          <span>{errorMessage}</span>
          <span>{userMsg}</span>
          <button className="loginButton" type='button' onClick={this.onSubmitHandler}>
            Login
          </button>

          <div className="login-signup-link">
            Dont have an account  
            <Link to="/signup">
              &nbsp;&nbsp; Sign Up
            </Link>
          </div>
        </form>
        
      </div>
    );
  }
}
const mapStateToProps = state => {
  const { isProcessing,  errorMessage, userMsg } = state.loginReducer;
  return {
    isProcessing, errorMessage, userMsg
  };
};
const mapDispatchToProps = dispatch => ({
  loggingIn: (email, password) => {
    dispatch(loginActions({ email, password }));
  }
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Login);
