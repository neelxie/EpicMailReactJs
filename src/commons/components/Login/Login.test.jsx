/* eslint-disable no-undef */
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Login from './index';
import { Provider } from 'react-redux';
import store from '../../../Mainstore/Store';

const props = {
    getArticles: jest.fn(),
    items: [{ slug: "slug" }]
  };
// test
describe('<Login> Component', () => {
  it('should match the snapshot', () => {
    const component = shallow(
    <Provider store={store}><Login {...props} /></Provider>);

    expect(toJson(component)).toMatchSnapshot();
  });
});
// describe('Login ', () => {
//   let wrapper;
//   const loginprops = {
//     handleSubmit: jest.fn(),
//     onChange: jest.fn(),
//     email: '',
//     isLoggingIn: false,
//     password: '',
//     errors: {},
//   };
//   it('renders the login page', () => {
//     wrapper = shallow(<Login {...loginprops} />);
//   });
//   it('matches snapshot', () => {
//     expect(toJson(wrapper)).toMatchSnapshot();
//   });
// });
// describe('login Component', () => {
//   const props = {
//     onChange: jest.fn(),
//     handleSubmit: jest.fn(),
//     checkValidEmail: jest.fn(),
//     propsLoginAction: jest.fn(),
//     isLoading: false,
//     isLoggingIn: false,
//     loginFailed: false,
//   };

//   const event = {
//     preventDefault: () => {},
//     target: {
//       name: 'email',
//       value: 'abc',
//     },
//   };
//   const wrapper = mount(<Login {...props} />);

//   it('matches snapshot', () => {
//     expect(toJson(wrapper)).toMatchSnapshot();
//   });

//   it('should call an onChange function', () => {
//     const component = shallow(
//       <Login {...props} />,
//     );
//     component.setState({ email: 'email' });
//     component.instance().onChange(event);
//     expect(component.instance().state.email).toBe('abc');
//   });

//   it('should call handleSubmit', () => {
//     const component = shallow(
//       <Login {...props} />,
//     );
//     component.setState({ email: 'jane@us.com', password: '' });
//     component.instance().handleSubmit(event);
//     expect(component.state('errors').userPassword).toBe('Password field is required');
//   });

//   it('should call a submit a form', () => {
//     wrapper.setState({
//       email: 'jennyj',
//     });
//     const userEmail = {
//       match: jest.fn(),
//     };
//     const instance = wrapper.instance();
//     const checkValidEmail = jest.spyOn(instance, 'checkValidEmail');
//     checkValidEmail(userEmail);
//     instance.handleSubmit({ preventDefault: jest.fn() });
//     expect(checkValidEmail).toHaveBeenCalled();
//   });

//   it('Valid Email', () => {
//     wrapper.setState({
//       email: 'jenny@gmail.com',
//     });
//     const userEmail = {
//       match: jest.fn(),
//     };
//     const instance = wrapper.instance();
//     const checkValidEmail = jest.spyOn(instance, 'checkValidEmail');
//     instance.handleSubmit({ preventDefault: jest.fn() });
//     checkValidEmail(userEmail);
//     expect(checkValidEmail).toHaveBeenCalled();
//   });

//   it('should call submit form', () => {
//     wrapper.setState({
//       email: 'jennyj',
//     });
//     const userEmail = {
//       match: jest.fn(),
//     };
//     const instance = wrapper.instance();
//     const checkValidEmail = jest.spyOn(instance, 'checkValidEmail');
//     instance.handleSubmit({ preventDefault: jest.fn() });
//     checkValidEmail(userEmail);
//     expect(checkValidEmail).toHaveBeenCalled();
//   });

//   it('should validate an empty password', () => {
//     const instance = wrapper.instance();
//     instance.handleSubmit({ preventDefault: jest.fn() });
//     expect('Password field is required').toEqual(instance.state.errors.userPassword);
//   });

//   it('should call loginAction when handling submit', () => {
//     wrapper.setState({
//       email: 'jennyzalwango12@gmail.com',
//       password: 'password',
//       errors: {},
//     });
//     const instance = wrapper.instance();
//     instance.handleSubmit({ preventDefault: jest.fn() });
//     props.propsLoginAction();
//     expect(props.propsLoginAction).toHaveBeenCalled();
//   });

//   it('should load the component', () => {
//     expect(wrapper).toHaveLength(1);
//   });
// });
