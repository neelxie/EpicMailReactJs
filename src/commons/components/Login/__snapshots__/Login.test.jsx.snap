// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login> Component should match the snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <Connect(Login)
    getArticles={[MockFunction]}
    items={
      Array [
        Object {
          "slug": "slug",
        },
      ]
    }
  />
</ContextProvider>
`;
